pkg/mpc-examples/contracts/03-ebu-rebalancer/EBURebalancerController.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/ILastCreatedPoolFactory.sol";
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/IManagedPool.sol";
+import "@balancer-labs/v2-interfaces/contracts/vault/IVault.sol";
+import "@balancer-labs/v2-pool-utils/contracts/lib/ComposablePoolLib.sol";
+
+// solhint-disable not-rely-on-time
+
+contract EBURebalancerController {
+    IVault private immutable _vault;
+    bytes32 private immutable _poolId;
+
+    uint256 private constant _MINIMUM_DURATION_BETWEEN_REBALANCE = 30 days;
+    uint256 private constant _REBALANCE_DURATION = 7 days;
+    uint256 private constant _MAX_SWAP_FEE_PERCENTAGE = 95e16; // 95%
+    uint256 private immutable _minSwapFeePercentage;
+    uint256 private _lastRebalanceCall;
+
+    event PoolRebalancing(uint256 indexed startTime, uint256 endTime);
+
+    constructor(IVault vault, uint256 minSwapFeePercentage) {
+        // Get poolId from the factory
+        bytes32 poolId = IManagedPool(ILastCreatedPoolFactory(msg.sender).getLastCreatedPool()).getPoolId();
+
+        // Verify that this is a real Vault and the pool is registered - this call will revert if not.
+        vault.getPool(poolId);
+
+        // Store Vault and poolId
+        _vault = vault;
+        _poolId = poolId;
+
+        _minSwapFeePercentage = minSwapFeePercentage;
+    }
+
+    function rebalancePool() public {
+        require(
+            block.timestamp - _lastRebalanceCall >= _MINIMUM_DURATION_BETWEEN_REBALANCE,
+            "Minimum time between calls not met"
+        );
+
+        if (isPoolPaused()) {
```

Comment:
This implies that we are fine with the pool not being paused to start a rebalance -- is that the intention? Should there be a required duration that the pool be paused before it is rebalanced?

Reply:
Yes that is the intention. What are your opinions about this? My assumption is that people would call this pause function immediately after rebalancing is complete

Reply:
If that assumption doesn't hold though, then calling `rebalancePool()` is kind of a waste of time. If someone doesn't pause the pool, then the pool will be at the nominal (~0%) swap fee, and should therefore already be balanced to market price, and bringing the swap fee up to 95% would then just make the pool less useful. I think it only makes sense to rebalance after the pool has been paused (or some other operation that could otherwise induce an instantaneous arbitrage opportunity)